client/src/hooks/store/useLeaderBoardStore.tsx

```diff
@@ -1,123 +1,234 @@
+
+import { Subscription } from "rxjs";
+import useBlockchainStore from "../store/useBlockchainStore";
+import { useContributions } from "../helpers/useContributions"
+import { ResourcesIds } from "@bibliothecadao/eternum";
+
+// Define elsewhere
+const PTS_PER_CYCLE = 100;
+const CYCLE_TIME = 24 * 60 * 60 * 1000;
+
+// Computed based on qty of 10 of each resource for hyperstructure building
+const TOTAL_EFFECTIVE_CONTRIBUTION = 8209.10;
+
+// fix ?
+const ResourceMultipliers: { [key in ResourcesIds]?: number } = {
+  [ResourcesIds.Wood]: 1.00,
+  [ResourcesIds.Stone]: 1.27,
+  [ResourcesIds.Coal]: 1.31,
+  [ResourcesIds.Copper]: 1.90,
+  [ResourcesIds.Obsidian]: 2.26,
+  [ResourcesIds.Silver]: 2.88,
+  [ResourcesIds.Ironwood]: 4.25,
+  [ResourcesIds.ColdIron]: 5.24,
+  [ResourcesIds.Gold]: 5.49,
+  [ResourcesIds.Hartwood]: 8.44,
+  [ResourcesIds.Diamonds]: 16.72,
+  [ResourcesIds.Sapphire]: 20.30,
+  [ResourcesIds.Ruby]: 20.98,
+  [ResourcesIds.DeepCrystal]: 20.98,
+  [ResourcesIds.Ignium]: 29.15,
+  [ResourcesIds.EtherealSilica]: 30.95,
+  [ResourcesIds.TrueIce]: 36.06,
+  [ResourcesIds.TwilightQuartz]: 45.18,
+  [ResourcesIds.AlchemicalSilver]: 53.92,
+  [ResourcesIds.Adamantine]: 91.20,
+  [ResourcesIds.Mithral]: 135.53,
+  [ResourcesIds.Dragonhide]: 217.92,
+  [ResourcesIds.Earthenshard]: 20.98,
+};
+
+export let TOTAL_CONTRIBUTABLE_AMOUNT: number;
+HYPERSTRUCTURE_TOTAL_COSTS_SCALED.forEach(({ resource, amount }) => {
+  TOTAL_CONTRIBUTABLE_AMOUNT += ResourceMultipliers[resource as keyof typeof ResourceMultipliers]! * amount;
+});
+
+function getResourceMultiplier(resourceType: BigInt): number {
+  const resourceTypeNumber: ResourcesIds = Number(resourceType);
+  return ResourceMultipliers[resourceTypeNumber] ?? 0;
+}
+
+function computeContributionPoints(totalPoints: number, qty: number, type: BigInt) {
+  const effectiveContribution = qty * getResourceMultiplier(type);
+  const points = effectiveContribution / TOTAL_EFFECTIVE_CONTRIBUTION * totalPoints;
+  return points;
+}
+
+export interface PlayerPointsLeaderboardInterface {
   address: string;
   addressName: string;
   order: string;
-  totalResources: number;
+  totalPoints: number;
   isYours: boolean;
 }
 
-export interface OrderResourceLeaderboardInterface {
-  order: string;
-  realmCount: number;
-  totalResources: number;
-  isYours: boolean;
-}
+// export interface OrderResourceLeaderboardInterface {
+//   order: string;
+//   realmCount: number;
+//   totalResources: number;
+//   isYours: boolean;
+// }
 
 interface LeaderboardStore {
   loading: boolean;
   setLoading: (loading: boolean) => void;
-  playerResourceLeaderboard: PlayerResourceLeaderboardInterface[];
-  orderResourceLeaderboard: OrderResourceLeaderboardInterface[];
-  setResourceLeaderboards: (
-    playerResourceLeaderboard: PlayerResourceLeaderboardInterface[],
-    orderResourceLeaderboard: OrderResourceLeaderboardInterface[],
+  playerPointsLeaderboard: PlayerPointsLeaderboardInterface[];
```

Comment:
do we need to have this in the state? Could we just fetch with recs?

Reply:
could be computation heavy to calculate leaderboard for all players so might be better to set it at the beginning and then have a refresh button, i think zustand store is good in this case